share/python/examples/gpu_viewer.py

```diff
@@ -0,0 +1,1367 @@
+            if default_cs:
+                default_cs_name = default_cs.getName()
+                if default_cs_name in all_cs_names:
+                    cs_idx = all_cs_names.index(default_cs_name)
+                    cs_actions[cs_idx].setChecked(True)
+
+            # Populate displays
+            for action in self.display_action_group.actions():
+                self.display_action_group.removeAction(action)
+            self.display_menu.clear()
+
+            displays = list(config.getDisplays())
+
+            for i, display in enumerate(displays):
+                action = self.display_menu.addAction(display)
+                action.setCheckable(True)
+                if i == 0:
+                    action.setChecked(True)
+                self.display_action_group.addAction(action)
+
+            default_display = config.getDefaultDisplay()
+
+            # Populate views. They will be re-populated whenever the display
+            # changes.
+            self.view_box.clear()
+            self.view_box.addItems(config.getViews(default_display))
+
+            default_view = config.getDefaultView(default_display)
+
+            # Set default display/view
+            self.set_display_view(default_display, default_view)
+
+            # Build OCIO processors
+            self.image_plane.update_ocio_proc(
+                input_cs=self.input_color_space(),
+                display=self.display(),
+                view=self.view()
+            )
+
+    @QtCore.Slot(QtWidgets.QAction)
+    def _on_input_colorspace_changed(self, action):
+        self.image_plane.update_ocio_proc(input_cs=action.text())
+
+    @QtCore.Slot(QtWidgets.QAction)
+    def _on_display_changed(self, action):
+        with self._ocio_signals_blocked():
+
+            config = ocio.GetCurrentConfig()
+            display = action.text()
+
+            # Re-populate views
+            current_view = self.view()
+            views = list(config.getViews(display))
+
+            self.view_box.clear()
+            self.view_box.addItems(views)
+
+            if current_view in views:
+                # Preserve current view if possible
+                view_idx = views.index(current_view)
+            else:
+                # Fallback on default if not
+                view_idx = self.view_box.findText(
+                    config.getDefaultView(display)
+                )
+
+            if view_idx != -1:
+                self.view_box.setCurrentIndex(view_idx)
+
+            self.image_plane.update_ocio_proc(display=display, view=self.view())
+
+    @QtCore.Slot(str)
+    def _on_view_changed(self, view):
+        self.image_plane.update_ocio_proc(view=view)
+
+    @QtCore.Slot(float)
+    def _on_exposure_changed(self, value):
+        self.image_plane.update_exposure(value)
+
+    @QtCore.Slot(float)
+    def _on_gamma_changed(self, value):
+        self.image_plane.update_gamma(value)
+
+
+if __name__ == "__main__":
```

Comment:
Because you need to be on the Core Profile on macOS to access the programmatic pipeline, I did update the default format by using:

```
format = QtOpenGL.QGLFormat()
format.setProfile(QtOpenGL.QGLFormat.CoreProfile)
format.setSampleBuffers(True)
format.setSwapInterval(1)
format.setVersion(3, 2)
QtOpenGL.QGLFormat.setDefaultFormat(format)
```

Might need a platform conditional if that breaks things on other platform, I didn't check.